src/graphics/vertex-iterator.js

```diff
@@ -1,39 +1,67 @@
 Object.assign(pc, function () {
     'use strict';
 
-    function VertexIteratorAccessor(buffer, vertexElement) {
+    function VertexIteratorAccessor(buffer, vertexElement, vertexFormat) {
         this.index = 0;
+        this.numComponents = vertexElement.numComponents;
+
+        // map only section of underlaying buffer for non-interleaved format
+        var bufferLength;
+        if (!vertexFormat.interleaved)
+            bufferLength = vertexFormat.vertexCount * vertexElement.numComponents;

```

Comment:
This must also account for elements requiring 4 byte boundary.

Reply:
call to pc.math.roundUp later in the function handles that bit - that rounds up the offset of start of data for each semantic.

Reply:
I'm not sure anything else is needed here. In case of non-interleaved format, the vertexElement.offset is already aligned when VertexFormat is created. As an example, if format is float, it will be 4 bytes aligned (size of component). bufferLength here is number of floats in that buffer.